### DISCUSSION TITLE
Add @export_if Tag for Conditional Property Exporting in GDScript/C#

### DISCUSSION THREAD

**Discussion by LouitLA** [godotengine/godot-proposals#10898] [Scripting]
**Feature Description:**

The proposed feature introduces a new `@export_if` tag for GDScript (and potentially C#) that allows for conditional exporting of variables in the Godot editor. This tag would enable developers to hide or show specific exported variables in the editor based on a boolean condition.

This is particularly useful when working with enumerations or toggle variables that determine which properties should be displayed. For example, if a property depends on an enum value, the `@export_if` tag would allow developers to conditionally export related properties, making the editor more intuitive and reducing clutter.

**How it Works:**

- _@export_if(condition):_ Export the following variables only if the condition evaluates to `true`. The condition could reference an exported enum or boolean variable.

- _@export_end:_ Marks the end of the block where the conditional export applies. All variables between `@export_if` and `@export_end` will only be visible in the editor if the condition is met.

**Benefits:**

- _Cleaner Editor Interface:_ Only show properties relevant to the current context, especially when using enums or feature toggles.

- _Improved Usability_: Reduces confusion by hiding irrelevant properties based on user selections.

- _Flexibility:_ Can be applied in various contexts like enums, boolean toggles, or even custom conditions.

**Motivation:**

The current workarounds to achieve a similar behavior in the Godot editor are not practical and involve unnecessary complexity. These alternative approaches not only require way more code but also make the script harder to maintain and less intuitive to use.

By introducing the `@export_if` feature, developers can minimize redundant code and focus on the core logic of their game or application. This will improve usability for both new and experienced Godot developers, making the editor more adaptable and responsive to the needs of their projects.

_Example:_

*Now:*
![image](https://github.com/user-attachments/assets/ea3e055a-c10f-443e-9726-8a7e8f7e594a)

*With This Feature:*
![image](https://github.com/user-attachments/assets/c59d4ab1-07c3-4791-94ff-7dd71e5ab726)

**Comment by AThousandShips** [comment]
There's already a proposal open for this:
* https://github.com/godotengine/godot-proposals/issues/9842
